src/pages/signup/StepThree.jsx

```diff
@@ -39,6 +39,59 @@ export default function StepThree(props) {
       [e.target.name]: e.target.value,
     }));
   };
+
+  /*
+    your next line is: "shouldn't we separate business logic from presentation layer?"
+    the answer is yes!
+    the available plans should be fetched from the backend
+    today I may be able loan 1000, tomorrow it could be 10000, interest rates change, etc.
+  */
+  function calculatePaymentSize(presentValue, numberOfPayments){
+    if(isNaN(presentValue) || presentValue <= 0 ||
+       isNaN(numberOfPayments) || numberOfPayments <= 0
+    ){
+      return 0
+    }
+    // 10% per year
+    const annualInterest = 0.1
+    // in some places you'll see annualInterest/12, annualInterest ^ (1/12) is also acceptable
+    const interest = (1 + annualInterest)**(1/12) - 1
+
+    let paymentSize =
+      presentValue * ((1+interest)**(numberOfPayments) * interest) /
+                      ((1+interest)**(numberOfPayments) -1)
+    // round it off
+    return paymentSize.toFixed(2)
+  }
+
+  // changing the amount after setting the number of  installments
+  const handleAmountChange = (e) =>{
+    let newAmount = parseInt(e.target.value)
+    let newInstallment = calculatePaymentSize(
+      newAmount,
+      values[SIGNUP_FIELDS.numberOfInstallments]
+    )
+    setValues({
+      ...values,
+      [SIGNUP_FIELDS.requestedLoanAmount] : newAmount,
+      [SIGNUP_FIELDS.installmentAmount]   : newInstallment
+    })
+  }
+
+  // changing the number of installments after setting the amount
+  const handleInstallmentChange = (e) =>{
+    let newInstallments = parseInt(e.target.value)
+    let newInstallment = calculatePaymentSize(
+      values[SIGNUP_FIELDS.requestedLoanAmount],
+      newInstallments
+    )
+    setValues({
+      ...values,
+      [SIGNUP_FIELDS.numberOfInstallments] : newInstallments,
+      [SIGNUP_FIELDS.installmentAmount]    : newInstallment
+    })
+  }
+
```

Comment:
nice! Let's move this logic out so it can be re-used in the admin / user portal as well